src/dotnet/App.Maui/Platforms/iOS/AppDelegate.cs

```diff
@@ -42,6 +52,28 @@ public override void DidEnterBackground(UIApplication application)
         base.DidEnterBackground(application);
     }
 
+    [Export ("messaging:didReceiveRegistrationToken:")]
+    public void DidReceiveRegistrationToken (Firebase.CloudMessaging.Messaging messaging, string fcmToken)
+    {
+        // Monitor token generation: To be notified whenever the token is updated.
+        var token = fcmToken;
+        Log.LogDebug("OnNewToken: '{Token}'", token);
+        var appServices = IPlatformApplication.Current?.Services;
+        if (appServices == null)
+            return;
+
+        var rpcHub = appServices.GetService<RpcHub>();
```

Comment:
I think we can extract code block below and use both from Android and iOS platforms.